src/client/Client.cc

```diff
@@ -2733,23 +2733,25 @@ void Client::handle_mds_map(const MConstRef<MMDSMap>& m)
       continue;  // no change
     
     session->mds_state = newstate;
-    if (old_inc != new_inc && newstate > MDSMap::STATE_RECONNECT) {
```

Comment:
how about changing this to
````   
 if ( newstate > MDSMap::STATE_RECONNECT &&
        (old_inc != new_inc || oldstate < MDSMap::STATE_RECONNECT ))
````

Reply:
It really does solve most cases but  i think this is still not completely safe. For example, if the send_connect request didn't be sent mds or for some other complex network reasons the PipeConnect didn't established successfully then it will lead this bug again because our oldstate is MDSMap::STATE_RECONNECT.

What do you think of this:
-      if (!(session->con->is_connected())) {
+      if (oldstate < MDSMap::STATE_RECONNECT || !(session->con->is_connected())) {

Reply:
I think messenger will make sure connection established. we don't not need to worry about connection

Reply:
Ok, thanks.